spec/lib/page_cloner_spec.rb

```diff
@@ -0,0 +1,111 @@
+require 'rails_helper'
+
+describe PageCloner do
+  let!(:tag)  { create(:tag) }
+  let(:campaign) { create(:campaign) }
+  let(:page)  { create(:page, tags: [tag], campaign: campaign) }
+  let!(:link) { create(:link, page: page) }
+
+  subject(:cloned_page) { PageCloner.clone!(page) }
+
+  it 'clones page' do
+    expect(cloned_page.id).not_to eq(page.id)
+  end
+
+  it 'clones links' do
+    expect(cloned_page.links).not_to eq(page.links)
+    expect(cloned_page.links.count).to eq(1)
```

Comment:
also this is unrelated to Tuuli's comment and it doesn't matter here but I think it's worth noting that you pretty much always want to use `.size` instead of `.count` because in ActiveRecord `count` makes a SQL `COUNT` query every time and size only does it if it's not already in memory. [More info](http://stackoverflow.com/questions/6083219/activerecord-size-vs-count)